ibmcloud/README.md

```diff
@@ -89,9 +89,16 @@ As usual, you need to create `terraform.tfvars` to specify parameter values. The
 
 ```
 ibmcloud_api_key = "<your API key>"
-ssh_key_name = "<your SSH key name>"
+user_ssh_key_name = "<your SSH key name>"
 cluster_name = "<cluster name>"
 ```
+Or
+```
+ibmcloud_api_key = "<your API key>"
+ssh_pub_key = "<your SSH public key>"
```

Comment:
With this approach we still have the problem that we need to use the `ssh_key_name` later and can't set it. Can we take take the approach of setting the key name always and then using the presence of absence of the `ssh_pub_key` field to tell whether to create a new one, or check an existing one (which I assume is the switch here?)

Reply:
I'm not sure what you mean by 
> we need to use the ssh_key_name later and can't set it

the rest of the terraform code needs the ssh ID which it either fetches from the data source, if provided a key_name or from the resource if provided a public key (which has created a new entry)

Reply:
The 'Launch cloud-api-adaptor` section requires the user to pass in the `ssh_key_name` in order to work. Currewntly the user would know the value of this as they specify it in the cluster creation, with this PR we don't let them set it, so are expecting them to have to look it up and haven't provided instructions for this? Does that make sense?

Reply:
Ah yes, thanks for pointing that out. I'd only been focused on the cluster creation section. Didn't realise it would be required elsewhere. Your earlier comment makes more sense now. I definitely think it should be possible to have the ssh_key_name be required and just switch the logic based upon the existance of the pub_key. I'll double check that and update the code with those changes. Thanks again.

Reply:
I would counter that point. If I get terraform to create and manage the key for me, why should I have to choose its name? If another programatic stage needs to know it [terraform output variables](https://www.terraform.io/language/values/outputs) are probably a better option.

Reply:
Or as the naming pattern is based upon the cluster-name variable could just update the reference to use that in the docs
i.e.
```
$ image_name=<pod VM image name>
$ ssh_key_name=<your cluster name>-ssh-key
$ vpc_name=tok-vpc
```
??

Reply:
@stevenhorsman This maybe going too far beyond the scope of this issue but as there are a lot of parameters that need extracting for launch step is it worth adding them directly to the output that Martin has mentioned. 
The step requires
- key-id
- image-id
- profile-name
- zone-name
- primary-subnet-id
- primary-security-group-id
- vpc-id
which could all be outputted by terraform?

Reply:
@jtumber-ibm - so after having a chat with Martin is seems like the best thing to do is create an output variable for the `ssh_key_name` in this step and longer-term is to use that as the input of the `Launch cloud-api-adaptor` submodule, once we've stitched all the terraform templates together.
Until we are in the state, in the short term, I think we should avoid trying to be too clever with defaults and just continue the pattern of prompting the user to enter the same value multiple times and then review and clean these all up later. Based on that my suggestion would just be to get the user to supply `ssh_key_name` always during the cluster creation and if they set `ssh_pub_key` use it as the name of the new key we generate.
Sorry to muck you around as we've been thinking and learning out loud. Does this make sense for the tactical step?

Reply:
Sure, I'll make those changes and update the documentation to mention the output variable too.

Reply:
> @stevenhorsman This maybe going too far beyond the scope of this issue but as there are a lot of parameters that need extracting for launch step is it worth adding them directly to the output that Martin has mentioned. The step requires
> 
> * key-id
> * image-id
> * profile-name
> * zone-name
> * primary-subnet-id
> * primary-security-group-id
> * vpc-id
>   which could all be outputted by terraform?

Sorry, I hadn't refreshed whilst typing, so hadn't see that message. I think your point matches my suggestion though - there is lots of room to improve the 'chaining' of variables, but that's outside the scope of this PR :)